contracts/tequila-train/io/src/lib.rs

```diff
@@ -191,58 +201,82 @@ pub struct GameLauncher {
 }
 
 impl GameLauncher {
-    fn assert_limit_range(maybe_limit: Option<u64>) {
+    fn check_limit_range(maybe_limit: Option<u64>) -> Result<(), Error> {
         if let Some(limit) = maybe_limit {
-            assert!((2..=8).contains(&limit));
+            if !(2..=8).contains(&limit) {
+                return Err(Error("The limit should lie in the range [2,8]".to_owned()));
```

Comment:
I prefer enum-type errors with `impl Display` somewhere (or even using `thiserror` if it is acceptable). It yields less allocations inside the functions.

```suggestion
                return Err(Error::WrongPlayersCount);
```